apps/src/templates/progress/progressTypes.js

```diff
@@ -88,3 +88,35 @@ export const lessonGroupType = PropTypes.shape({
   bigQuestions: PropTypes.string,
   description: PropTypes.string
 });
+
+/**
+ * @typedef {Object} StudentLevelProgress
+ *
+ * @property {string} status
+ * A string enum representing student progress status on a level.
+ * See src/util/sharedConstants.LevelStatus.
+ * @property {number} result
+ * A numerical enum of the TestResult a student received for a level.
+ * See src/constants.TestResult.
+ * See src/code-studio/activityUtils.activityCssClass for a mapping to status.
+ * @property {bool} paired
+ * A boolean indicating if a student was paired on a level.
+ * @property {number} timeSpent
+ * An optional value indicating the time a student spent on a level.
+ * @property {array} pages
+ * Array of StudentLevelProgress objects representing progress on individual
+ * pages of a multi-page assessment
+ */
+const studentLevelProgressShape = {
+  status: PropTypes.string.isRequired,
+  result: PropTypes.number.isRequired,
```

Comment:
Suggestion: I think renaming `result` to `testResult` (and all of the functions that handle this field might help clarify the difference between `status` and `result`)?  Up to you and Charlie.

Reply:
that's a great idea, but we currently use "result" all over our frontend and backend code, so i think that is out of scope for this work.